lib/Doctrine/Persistence/Mapping/Driver/FileDriver.php

```diff
@@ -32,11 +34,10 @@ abstract class FileDriver implements MappingDriver
      * Initializes a new FileDriver that looks in the given path(s) for mapping
      * documents and operates in the specified operating mode.
      *
-     * @param string|string[]|FileLocator $locator       A FileLocator or one/multiple paths
-     *                                                where mapping documents can be found.
-     * @param string|null                 $fileExtension
+     * @param string|string[]|FileLocator $locator A FileLocator or one/multiple paths
+     *                                          where mapping documents can be found.
      */
-    public function __construct($locator, $fileExtension = null)
+    public function __construct($locator, ?string $fileExtension = null)
```

Comment:
I don't think other classes (e.g. `PHPDriver`) constructors _should_ automatically inherit these annotations, as constructor calls aren't subservient to the Liskov Substitution Principle. So I think these annotations need to be on every subclass with its own constructor.

Reply:
I will fixup the constructors. I see what you mean now that I look at it.